File: frontend/src/api/dto/updateUserRequest.ts
import type { UnitSystem } from './unitSystem.ts'

/**
 * User profile update request
 */
export interface UpdateUserRequest {
  /**
   * User display name
   * @minLength 1
   * @maxLength 200
   */
  displayName?: string
  /** Preferred unit system */
  unitSystem?: UnitSystem
}

File: backend/src/main/java/fr/pedalons/dto/users/request/UpdateUserRequest.java
package fr.pedalons.dto.users.request;

import fr.pedalons.dto.validation.ValidateSchema;
import fr.pedalons.enums.UnitSystem;
import jakarta.validation.constraints.Size;
import lombok.Builder;
import org.eclipse.microprofile.openapi.annotations.media.Schema;
import org.jspecify.annotations.Nullable;

@Schema(description = "User profile update request")
@ValidateSchema
@Builder
public record UpdateUserRequest(
    @Nullable
        @Schema(description = "User display name", examples = "John Doe")
        @Size(min = 1, max = 200)
        String displayName,
    @Nullable @Schema(description = "Preferred unit system", examples = "metric")
        UnitSystem unitSystem) {}
